server/models/user.ts

```diff
@@ -302,20 +296,11 @@ userSchema.statics.findByEmailOrUsername = function findByEmailOrUsername(
   return this.findByUsername(value, callback);
 };
 
-/**
- *
- * Queries User collection, performing a MongoDB logical or with the email
- * and username (i.e. if either one matches, will return the first document).
- *
- * @param {string} email
- * @param {string} username
- * @callback [cb] - Optional error-first callback that passes User document
- * @return {Promise<Object>} - Returns Promise fulfilled by User document
- */
```

Comment:
We do want to keep a JSDoc block to describe the function, if one exists already.  In this case the description of the function is helpful because it describes what the function does.

I'm not certain of the best way to handle the descriptions of the parameters.  I think the most common is to keep the description text but remove the types.  We can move those comments inline between the parameters but I'm not sure if that gets the same intellisense support?  I have to play.  It's also possible that we don't really need them in the majority of cases because they don't add any information that's not already in your new types.  Like "Optional error-first callback that passes User document" doesn't tell us anything that `cb?: (err: Error | null, doc?: User) => void` doesn't.  The types indicate that `cb` is optional, the error is the first argument, the value is a `User`, and more.